### DISCUSSION TITLE
alert api incomplete results

### DISCUSSION THREAD

**Discussion by cc8080** [weather-gov/api#469] [Q&A]
I want to get all active alerts that impact a particular point.  But, my call to the alerts endpoint isn't showing all active alerts.

Right now, this API call returns one "feature":

- URL: `https://api.weather.gov/alerts/active?point=27.4992,-80.3422`

- Returns: Rip Current Statement

But, the mobile site shows two alerts:

- URL: `https://mobile.weather.gov/index.php?lat=27.4992&lon=-80.3422`
- Returns: Rip Current Statement and Hazardous Weather Outlook
  
The mobile site is loading the hazards from this URL:

- `https://mobile.weather.gov/services/wwaShowJSON.php?warnzone=FLZ059&warncounty=FLC111&firewxzone=FLZ059&local_place1=Saint+Lucie+FL&zinfo=E%7CY%7C5`

So, it appears to be pulling information from the zone and county data. I was under the impression that the API point endpoint would include anything that affects the point, regardless of the source.

Please advise. Thanks!

**Comment by jah1007** [comment]
Hazardous Weather Outlooks were purposefully dropped from the upstream source of data for the API. They generally are seen in the Weather Service as forecasts and not alerts which is ultimately why they were not carried forward. mobile.weather.gov uses a different data source than the API and is tied to an older, legacy data source.

**Comment by scadergit** [comment]
To that point, the mobile weather site itself will be sunset, soon.

**Comment by JJKraw** [comment]
Note that there was a known bug where alerts for fire zones, when the fire zone did not have the same ID as the forecast zone, were not returned for a point query. The last update I got from NCO Ops was that a fix was to deploy sometime back in July but only now did I think to go back and try to test it. And, of course, there are no fire type alerts active.

**Reply by jah1007** [reply]
I think that one may be resolved with the latest software release that was implemented this week but I would need @StephenClouse to confirm that one.

**Comment by cc8080** [comment]
Thanks for the info. It looks like the way to get "hazards" is only through the "raw" results from `https://api.weather.gov/gridpoints/{office}/{grid X},{grid Y}`.  But that's the same endpoint that's frequently giving me stale data. Is there another way through the new API that doesn't have the stale data problem?
Thanks.

**Reply by JJKraw** [reply]
Not the answer to your question, but the stale data issue is known (has to do with caching I believe). The NWS guys can fill in on where that's at.

**Comment by cc8080** [comment]
Yes, I'm following the stale data issue. I'm trying to find a workaround until it gets fixed. And I'm having a hard time figuring out if the data I'm pulling is "complete". If parts of the earlier (DWML) data have been "dropped and not replaced", and the data I can get is inconsistent with what the current website shows, that's very confusing for me and for users.

So, I think my question boils down to: If I just get the alerts from api.weather.gov/alerts, am I missing important  information?

**Reply by scadergit** [reply]
That's a question for your local weather office, as it's more policy than system design. Their contact is at the bottom of each local office page on weather.gov. The current version of alerts (alerts.weather.gov) generates alerts from text products--lots of policy at the technical level. In contrast, the API is fed from upstream, which now does the translation from text product to CAP, and they have determined to exclude HWO and such. The roadmap will eventually move the effort to the local offices who will create CAP directly.

**Reply by jah1007** [reply]
So the Hazard grid you are attempting to pull from within the forecast data will not contain all alerts found on the alerts endpoint. This is because while the alerts come from one forecast office they are generated by more than one application at that office. The hazard grid just represents data from one of those applications. These grids will not contain important alerts such as Tornado or Flash Flood Warnings. I would recommend against gathering alerts from the forecast grids.
